Gems/PhysX/Code/Source/MeshColliderComponent.cpp

```diff
@@ -79,6 +79,11 @@ namespace PhysX
         {
             AZ::Data::AssetBus::MultiHandler::BusConnect(m_shapeConfiguration->m_asset.GetId());
             m_shapeConfiguration->m_asset.QueueLoad();
+
+            // The asset loading must be completed because the shapes are required to be ready
+            // at activation time, so the rigid body can query them when OnEntityActivated
+            // event is called.
+            m_shapeConfiguration->m_asset.BlockUntilLoadComplete();
```

Comment:
question: I know the advise is generally to avoid blocking loads as much as possible. Might be worth checking if there's another solution (tagging @mbalfour-amzn, @AMZN-stankowi and @amzn-mike in case they have any ideas). It might be okay for now but I just would like to check).

Reply:
This is a issue from how we are initializing colliders, there is nothing related to the asset pipeline that can help us here. This is the first out of scope point of the description:

> - Improve MeshCollider component to not block loading its asset. This can be done in the future by making colliders notify to rigid bodies of new/modified shapes and rigid bodies to be able to listen to those notifications and recreate themselves when that happens, but that's out of scope for this work.

But that's not a trivial change to make and better to cover it on its own in the future.

Reply:
Okay understood, thanks for clarifying - would definitely be good to return to at some point in future but agree now is not the time :+1: